Add docs for `HTMLScriptElement.async` and `HTMLScriptElement.defer` (#33286)
mdn/content: files/en-us/web/api/htmlscriptelement/async/index.md

---
title: "HTMLScriptElement: async property"
short-title: async
slug: Web/API/HTMLScriptElement/async
page-type: web-api-instance-property
browser-compat: api.HTMLScriptElement.async
---

{{APIRef("HTML DOM")}}

> The **`async`** property of the {{domxref("HTMLScriptElement")}} interface is a boolean value that controls how the script should be executed. For classic scripts, if the `async` property is set to `true`, the script (the `src` property must be specified) will be fetched in parallel to parsing and evaluated as soon as it is available. For [module scripts](/en-US/docs/Web/JavaScript/Guide/Modules), if the `async` property is set to `true`, the script and all their dependencies will be fetched in parallel to parsing and evaluated as soon as they are available.

Author: ```suggestion
The **`async`** property of the {{domxref("HTMLScriptElement")}} interface is a boolean value that controls how the script should be executed. For classic scripts, if the `async` property is set to `true`, the external script will be fetched in parallel to parsing and evaluated as soon as it is available. For [module scripts](/en-US/docs/Web/JavaScript/Guide/Modules), if the `async` property is set to `true`, the script and all their dependencies will be fetched in parallel to parsing and evaluated as soon as they are available.
```

---
title: "HTMLScriptElement: async property"
short-title: async
slug: Web/API/HTMLScriptElement/async
page-type: web-api-instance-property
browser-compat: api.HTMLScriptElement.async
---

{{APIRef("HTML DOM")}}

> The **`async`** property of the {{domxref("HTMLScriptElement")}} interface is a boolean value that controls how the script should be executed. For classic scripts, if the `async` property is set to `true`, the script (its `src` property must be specified) will be fetched in parallel to parsing and evaluated as soon as it is available. For [module scripts](/en-US/docs/Web/JavaScript/Guide/Modules), if the `async` property is set to `true`, the script and all their dependencies will be fetched in parallel to parsing and evaluated as soon as they are available.

Author: ```suggestion
The **`async`** property of the {{domxref("HTMLScriptElement")}} interface is a boolean value that controls how the script should be executed. For classic scripts, if the `async` property is set to `true`, the script (the `src` property must be specified) will be fetched in parallel to parsing and evaluated as soon as it is available. For [module scripts](/en-US/docs/Web/JavaScript/Guide/Modules), if the `async` property is set to `true`, the script and all their dependencies will be fetched in parallel to parsing and evaluated as soon as they are available.
```

> 
> The `async` property takes higher priority than the `defer` property, which means that, if the two properties both specified, the `async` property will take effect.
> 
> The `defer` property may be specified with the `async` property, so legacy browsers that only support `defer` (and not `async`) fall back to the `defer` behavior instead of the default blocking behavior.
> 
> It reflects the `async` attribute of the {{HTMLElement("script")}} element.

Author: ```suggestion

It reflects the `async` attribute of the {{HTMLElement("script")}} element.
```
